pysoa/client/client.py

```diff
@@ -134,6 +137,9 @@ def send_request(self, job_request, message_expiry_in_seconds=None):
             self.metrics.commit()
 
     def _get_response(self, receive_timeout_in_seconds=None):
+        if self.unexpected_responses:
+            return self.unexpected_responses.popitem(last=False)
```

Comment:
if there is an unexpected_response, that never gets consumed, then this is going to keep on popping the same one..

Reply:
It'll only "never get consumed" by clients calling `get_response_for_request`, which is to be expected.